### PR TITLE
Mn/setdatabase

### PR DESCRIPTION
**Pull request by matydn05**
-Installed dependencies such as nmp, sequelize, sequelizecli, pg..
-Modified and added code to integrate PostgreSQL and test the Database.

### CODE REVIEW

File: src/config/config.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "development": {
+    "dialect": "postgres",
+    "host": "127.0.0.1",
+    "database": "chatdatabase",
+    "username": "maty",
```
Comment (abelosorio) [inline src/config/config.json]: Even though this is a sample project, you shouldn't use your name as the db's username. We usually use the same name for db name and username. In your case `chatdatabase`.

File: src/migrations/20190508114210-create-user.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    return queryInterface.createTable('users', {
+      id: {
+        type: DataTypes.UUID,
+        primaryKey: true,
+        defaultValue: DataTypes.UUIDV4
+      },
+      username: {
+        type: DataTypes.STRING,
+        allowNull: false,
+        unique: true
```
Comment (abelosorio) [inline src/migrations/20190508114210-create-user.js:12]: :+1:

File: src/migrations/20190508130329-create-salt.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        type: DataTypes.STRING,
+        allowNull: false
+      },
+      createdAt: {
+        type: DataTypes.DATE,
+        allowNull: false
+      },
+      updatedAt: {
+        type: DataTypes.DATE,
+        allowNull: false
```
Comment (abelosorio) [inline src/migrations/20190508130329-create-salt.js]: `updatedAt` can be null.

File: src/models/user.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    lastName: {
+      type: DataTypes.STRING,
+      allowNull: false
+    },
+    password: {
+      type: DataTypes.STRING,
+      allowNull: false
+    }
+  }, {})
```
Comment (abelosorio) [inline src/models/user.js]: The empty object at the end is not necessary.

File: src/testModels.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import models from './models'
```
Comment (abelosorio) [inline src/testModels.js]: You shouldn't commit test files (tests that you use for your own).

File: src/config/config.json (hunk 1 above)
Comment (matydn05) [reply src/config/config.json]: Fixed

File: src/migrations/20190508130329-create-salt.js (hunk 3 above)
Comment (matydn05) [reply src/migrations/20190508130329-create-salt.js]: Fixed

File: src/models/user.js (hunk 4 above)
Comment (matydn05) [reply src/models/user.js]: Fixed

File: src/testModels.js (hunk 5 above)
Comment (matydn05) [reply src/testModels.js]: Test file removed.

**Comment by matydn05** [comment]
Conflicts solved, ready to merge.

**Review by abelosorio — APPROVED** [review]
:+1: nice work!

File: src/models/salt.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      allowNull: false
+    }
+  })
+  // Class methods
+  Salt.getPasswordSalt = async function () {
+    const lastSalt = await this.getLast()
+
+    if (lastSalt) return lastSalt
+
+    return this.createRandom()
```
Comment (mpautasso) [inline src/models/salt.js]: We can join lines 19 and 21 in a single line doing `return lastSalt || this.createRandom()`

**Review by mpautasso — CHANGES_REQUESTED** [review]
@matydn05 only a very small change, nice work!

**Review by abelosorio — APPROVED** [review]
:+1:

File: src/models/salt.js (hunk 6 above)
Comment (matydn05) [reply src/models/salt.js]: Done !

**Pull request merged**